app/services/bibid_updater.rb

```diff
@@ -0,0 +1,78 @@
+# frozen_string_literal: true
+
+# Service to update Voyager bibids to alma mms ids
+class BibidUpdater
+  def self.update(logger: Logger.new(STDOUT), batch_size: 1000)
+    new(
+      query_service: Valkyrie::MetadataAdapter.find(:postgres).query_service,
+      logger: logger,
+      batch_size: batch_size
+    ).update
+  end
+
+  attr_reader :query_service, :logger, :batch_size
+  delegate :run_query, to: :query_service
+  def initialize(query_service:, logger:, batch_size: 1000)
+    @query_service = query_service
+    @logger = logger
+    @batch_size = batch_size
+  end
+
+  def update
+    logger.info "Updating #{total_count} records"
+    progress_bar
+    (1..iterations).each do
+      run_query(update_query)
+      progress_bar.progress += 1
+    end
+
+    Reindexer.reindex_all
```

Comment:
Suggestion: I'd only do reindex_works. FileSets don't have source_metadata_identifiers. In fact, I wonder if there's a way to get the ones that have changed?